web/satellite/src/components/project/ProjectDetails.vue

```diff
@@ -0,0 +1,285 @@
+					<HeaderedInput
+						label="Description"
+						placeholder ="Enter Description"
+						width="70vw"
+						height="10vh"
+						isMultiline
+						@setData="setNewDescription" />
+					<div class="project-details-info-container__description-container__buttons-area">
+						<Button label="Cancel" width="180px" height="48px" :onPress="toggleEditing" isWhite/>
+						<Button label="Save" width="180px" height="48px" :onPress="onSaveButtonClick"/>
+					</div>
+				</div>
+			</div>
+			<div class="project-details__button-area" id="deleteProjectPopupButton">
+				<Button class="delete-project" label="Delete project" width="180px" height="48px" :onPress="toggleDeleteDialog" isDeletion/>
+			</div>
+		</div>
+		<DeleteProjectPopup v-if="isPopupShown" />
+	</div>
+</template>
+
+<script lang="ts">
+    import { Component, Vue } from 'vue-property-decorator';
+    import Button from '@/components/common/Button.vue';
+    import HeaderedInput from '@/components/common/HeaderedInput.vue';
+    import Checkbox from '@/components/common/Checkbox.vue';
+    import EmptyState from '@/components/common/EmptyStateArea.vue';
+    import { PROJETS_ACTIONS, APP_STATE_ACTIONS, NOTIFICATION_ACTIONS } from '@/utils/constants/actionNames';
+    import ROUTES from '@/utils/constants/routerConstants';
+    import DeleteProjectPopup from '@/components/project/DeleteProjectPopup.vue';
+    import ProjectNavigation from '@/components/project/ProjectNavigation.vue';
+
+    @Component(
+        {
+            data: function () {
+                return {
+                    isEditing: false,
+                    newDescription: '',
+                };
+            },
+            methods: {
+                toggleEditing: function (): void {
+                    this.$data.isEditing = !this.$data.isEditing;
+                    // TODO: cache this value in future
+                    this.$data.newDescription = '';
+                },
+                setNewDescription: function (value: string): void {
+                    this.$data.newDescription = value;
+                },
+                onSaveButtonClick: async function (): Promise<any> {
+                    let response = await this.$store.dispatch(
+                        PROJETS_ACTIONS.UPDATE, {
+                            id: this.$store.getters.selectedProject.id,
+                            description: this.$data.newDescription,
+                        }
+                    );
+
+                    response.isSuccess
+                        // TODO: call toggleEditing method instead of this IIF
+                        ? (() => {
+                            this.$data.isEditing = !this.$data.isEditing;
+                            this.$data.newDescription = '';
+                            this.$store.dispatch(NOTIFICATION_ACTIONS.SUCCESS, 'Project updated successfully!');
+                        })()
```

Comment:
Do we really need ternary expression here?
I think it will be more readable with regular if/else